src/vs/code/node/cli.ts

```diff
@@ -319,23 +319,89 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (platform() !== 'darwin') {
+			child = spawn(process.execPath, argv.slice(2), options);
 
-		if (args.wait && waitMarkerFilePath) {
-			return new Promise<void>(resolve => {
+			if (args.wait && waitMarkerFilePath) {
+				return new Promise<void>(resolve => {
 
-				// Complete when process exits
-				child.once('exit', () => resolve(undefined));
+					// Complete when process exits
+					child.once('exit', () => resolve(undefined));
 
-				// Complete when wait marker file is deleted
-				whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
-			}).then(() => {
+					// Complete when wait marker file is deleted
+					whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
+				}).then(() => {
 
-				// Make sure to delete the tmp stdin file if we have any
-				if (stdinFilePath) {
-					unlinkSync(stdinFilePath);
+					// Make sure to delete the tmp stdin file if we have any
+					if (stdinFilePath) {
+						unlinkSync(stdinFilePath);
+					}
+				});
+			}
+		} else {
+			const requiresWait = args['status'] || args['verbose'] || hasReadStdinArg;
+			const envVars: string[] = [];
+			for (const envKey in env) {
+				if (envKey === '_') {
```

Comment:
This would benefit from a comment as to why `_` is filtered out.

Reply:
But should we warn to the console in this case to inform the user? Or can this not happen in reality.

Reply:
The loop isn't need now for the code-cli.sh approach.